places/views.py

```diff
@@ -0,0 +1,55 @@
+from django.http import JsonResponse
+import requests, json, os
+from dotenv import load_dotenv
+
+# reading .env file
+load_dotenv()
+
+
+def index(request): 
+    q = request.GET.get("q", False)
```

Comment:
There is a lot to unpack here, you can either search for places or lat and long, your implementation here is saying you can search for both


Even if that's the case, you didn't take care of when they send empty location

So this function should have been like this

```
       q = request.GET.get("q", '')
        lng = request.GET.get("lng", '')
        lat = request.GET.get("lat", '')
        API_KEY = os.getenv('API_KEY')
        searchUrl = "https://maps.googleapis.com/maps/api/place/findplacefromtext/json?inputtype=textquery&"
         searchResponse = requests.get(f'{searchUrl}input={q}&locationbias=point:{lng},{lat}&key={API_KEY}') 
        ```

And there shouldn't be anything wrong with it

Reply:
Google API expects places. You can decide to include the lat and long for more specific result. It's not either places or lat/lng

I handled empty location

Did you just change `False` to `''`?

Reply:
Yes, they are both falsy values

Reply:
So either is good?
Or is `''` better?

I'm tryna understand the reason behind the change ni o

Reply:
so you don't need to use an if anywhere, `''` already satisfies that if it's not required its a wild card search

Reply:
But I need the if to handle empty or False search